From 18480fd20b967d2be566c1d0d4b0622da842e18e Mon Sep 17 00:00:00 2001
From: "shoyeb.khan" <shoyeb.khan@lge.com>
Date: Thu, 2 Dec 2021 11:55:45 +0530
Subject: [PATCH] Fix provision failed issue

:Release Notes:
Fix provision failed issue

:Detailed Notes:
When a node is created/provisioned/imported,
the new node's agent is removed.
The corresponding pointer in the node structure needs
to be set to NULL to avoid pointing to freed memory.

:Testing Performed:
Built and tested

:QA Notes:
NA

:Issues Addressed:
[WRN-11804] [webOS OSE]: BLE Mesh Provision
             is not happening.

Upstream-Status: Inappropriate[webos specific]
---
 mesh/node.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/mesh/node.c b/mesh/node.c
index f7ac5ff..079ad9a 100644
--- a/mesh/node.c
+++ b/mesh/node.c
@@ -1121,6 +1121,8 @@ static void convert_node_to_storage(struct mesh_node *node,
 {
 	const struct l_queue_entry *entry;
 
+	memset(db_node, 0, sizeof(struct mesh_config_node));
+
 	db_node->cid = node->comp.cid;
 	db_node->pid = node->comp.pid;
 	db_node->vid = node->comp.vid;
@@ -2325,7 +2327,7 @@ void node_finalize_new_node(struct mesh_node *node, struct mesh_io *io)
 
 	free_node_dbus_resources(node);
 	mesh_agent_remove(node->agent);
-
+	node->agent = NULL;
 	node->busy = false;
 
 	/* Register callback for the node's io */
-- 
2.17.1

